File: Python3/231. Power of Two.py
class Solution:
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        a = 1
        while True:
            if n == a:
                return True
            if n < a:
                return False
            a = a * 2


sol = Solution()
for i in range(100000):
    if sol.isPowerOfTwo(i):
        print(i)

File: 231. Power of Two.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n <= 0) return false;
        return (n & (n - 1)) == 0;
    }
};

int main() {
    cout << Solution().isPowerOfTwo(3) << endl;
    cout << Solution().isPowerOfTwo(16) << endl;
    return 0;
}
